File: miniecs.py
import socket
import sys
import os

def createContainer(name):
    # Create a TCP/IP socket
    sock = socket.create_connection(('localhost', 7070))

    try:
            mesStr = "create, {0}".format(name)
            message = bytes(mesStr, 'utf-8')
            print('sending {!r}'.format(message))
            sock.sendall(message)

            amount_received = 0
            amount_expected = 6

            while amount_received < amount_expected:
                data = sock.recv(160)
                amount_received += len(data)
                print('received {!r}'.format(data))

    finally:
        print('closing socket')
        sock.close()
        return "creation successful"

def startContainer(name):
    # Create a TCP/IP socket
    sock = socket.create_connection(('localhost', 7070))

    try:
            mesStr = "start, {0}".format(name)
            message = bytes(mesStr, 'utf-8')
            print('sending {!r}'.format(message))
            sock.sendall(message)

            amount_received = 0
            amount_expected = 6

            while amount_received < amount_expected:
                data = sock.recv(160)
                amount_received += len(data)
                print('received {!r}'.format(data))

    finally:
        print('closing socket')
        sock.close()
        return "start successful"

def stopContainer(name):
    # Create a TCP/IP socket
    sock = socket.create_connection(('localhost', 7070))

    try:
            mesStr = "stop, {0}".format(name)
            message = bytes(mesStr, 'utf-8')
            print('sending {!r}'.format(message))
            sock.sendall(message)

            amount_received = 0
            amount_expected = 6

            while amount_received < amount_expected:
                data = sock.recv(160)
                amount_received += len(data)
                print('received {!r}'.format(data))

    finally:
        print('closing socket')
        sock.close()
        return "stop successful"

def deleteContainer(name):
    # Create a TCP/IP socket
    sock = socket.create_connection(('localhost', 7070))

    try:
            mesStr = "delete, {0}".format(name)
            message = bytes(mesStr, 'utf-8')
            print('sending {!r}'.format(message))
            sock.sendall(message)

            amount_received = 0
            amount_expected = 6

            while amount_received < amount_expected:
                data = sock.recv(160)
                amount_received += len(data)
                print('received {!r}'.format(data))

    finally:
        print('closing socket')
        sock.close()
        return "deletion successful"

def listContainers():
    # Create a TCP/IP socket
    sock = socket.create_connection(('localhost', 7070))

    try:
            mesStr = "list"
            message = bytes(mesStr, 'utf-8')
            print('sending {!r}'.format(message))
            sock.sendall(message)

            amount_received = 0
            amount_expected = 6

            while amount_received < amount_expected:
                data = sock.recv(160)
                amount_received += len(data)
                print('received {!r}'.format(data))

    finally:
        print('closing socket')
        sock.close()
        return 0

File: miniecs.cpp
#include <stdio.h>	//printf
#include <string.h>	//strlen
#include <sys/socket.h>	//socket
#include <arpa/inet.h>	//inet_addr
#include <unistd.h>
#include <string>

int createContainer(string name) 
{
	int sock;
	struct sockaddr_in server;
	char message[1000] , server_reply[2000];
	
	//Create socket
	sock = socket(AF_INET , SOCK_STREAM , 0);
	if (sock == -1) 
    {
		printf("Could not create socket");
	}
	puts("Socket created");
	
	server.sin_addr.s_addr = inet_addr("127.0.0.1");
	server.sin_family = AF_INET;
	server.sin_port = htons(7070);

	//Connect to remote server
	if (connect(sock , (struct sockaddr *)&server , sizeof(server)) < 0) 
    {
		perror("connect failed. Error");
		return 1;
	}
	
	puts("Connected\n");
	
    string mesAux = "create" + name;
    strcpy(message, mesAux); 
    
    //Send some data
    if( send(sock , message , strlen(message) , 0) < 0) 
    {
        puts("Send failed");
        return 1;
    } 
    else 
    {
        puts("send ok");
    }
    
    //Receive a reply from the server
    memset ( server_reply, 0, 2000 );
    if( recv(sock , server_reply , 2000 , 0) < 0) 
    {
        puts("recv failed");
        break;
    } 
    else 
    {
        puts("recv ok");
    }
    
    puts("Server reply :");
    puts(server_reply);
	
	close(sock);
	return 0;
}

int startContainer(string name)
{
    int sock;
    struct sockaddr_in server;
    char message[1000], server_reply[2000];

    // Create socket
    sock = socket(AF_INET, SOCK_STREAM, 0);
    if (sock == -1)
    {
        printf("Could not create socket");
    }
    puts("Socket created");

    server.sin_addr.s_addr = inet_addr("127.0.0.1");
    server.sin_family = AF_INET;
    server.sin_port = htons(7070);

    // Connect to remote server
    if (connect(sock, (struct sockaddr *)&server, sizeof(server)) < 0)
    {
        perror("connect failed. Error");
        return 1;
    }

    puts("Connected\n");

    string mesAux = "start" + name;
    strcpy(message, mesAux);

    // Send some data
    if (send(sock, message, strlen(message), 0) < 0)
    {
        puts("Send failed");
        return 1;
    }
    else
    {
        puts("send ok");
    }

    // Receive a reply from the server
    memset(server_reply, 0, 2000);
    if (recv(sock, server_reply, 2000, 0) < 0)
    {
        puts("recv failed");
        break;
    }
    else
    {
        puts("recv ok");
    }

    puts("Server reply :");
    puts(server_reply);

    close(sock);
    return 0;
}

int stopContainer(string name)
{
    int sock;
    struct sockaddr_in server;
    char message[1000], server_reply[2000];

    // Create socket
    sock = socket(AF_INET, SOCK_STREAM, 0);
    if (sock == -1)
    {
        printf("Could not create socket");
    }
    puts("Socket created");

    server.sin_addr.s_addr = inet_addr("127.0.0.1");
    server.sin_family = AF_INET;
    server.sin_port = htons(7070);

    // Connect to remote server
    if (connect(sock, (struct sockaddr *)&server, sizeof(server)) < 0)
    {
        perror("connect failed. Error");
        return 1;
    }

    puts("Connected\n");

    string mesAux = "stop" + name;
    strcpy(message, mesAux);

    // Send some data
    if (send(sock, message, strlen(message), 0) < 0)
    {
        puts("Send failed");
        return 1;
    }
    else
    {
        puts("send ok");
    }

    // Receive a reply from the server
    memset(server_reply, 0, 2000);
    if (recv(sock, server_reply, 2000, 0) < 0)
    {
        puts("recv failed");
        break;
    }
    else
    {
        puts("recv ok");
    }

    puts("Server reply :");
    puts(server_reply);

    close(sock);
    return 0;
}

int deleteContainer(string name)
{
    int sock;
    struct sockaddr_in server;
    char message[1000], server_reply[2000];

    // Create socket
    sock = socket(AF_INET, SOCK_STREAM, 0);
    if (sock == -1)
    {
        printf("Could not create socket");
    }
    puts("Socket created");

    server.sin_addr.s_addr = inet_addr("127.0.0.1");
    server.sin_family = AF_INET;
    server.sin_port = htons(7070);

    // Connect to remote server
    if (connect(sock, (struct sockaddr *)&server, sizeof(server)) < 0)
    {
        perror("connect failed. Error");
        return 1;
    }

    puts("Connected\n");

    string mesAux = "delete" + name;
    strcpy(message, mesAux);

    // Send some data
    if (send(sock, message, strlen(message), 0) < 0)
    {
        puts("Send failed");
        return 1;
    }
    else
    {
        puts("send ok");
    }

    // Receive a reply from the server
    memset(server_reply, 0, 2000);
    if (recv(sock, server_reply, 2000, 0) < 0)
    {
        puts("recv failed");
        break;
    }
    else
    {
        puts("recv ok");
    }

    puts("Server reply :");
    puts(server_reply);

    close(sock);
    return 0;
}

int listContainers(string name)
{
    int sock;
    struct sockaddr_in server;
    char message[1000], server_reply[2000];

    // Create socket
    sock = socket(AF_INET, SOCK_STREAM, 0);
    if (sock == -1)
    {
        printf("Could not create socket");
    }
    puts("Socket created");

    server.sin_addr.s_addr = inet_addr("127.0.0.1");
    server.sin_family = AF_INET;
    server.sin_port = htons(7070);

    // Connect to remote server
    if (connect(sock, (struct sockaddr *)&server, sizeof(server)) < 0)
    {
        perror("connect failed. Error");
        return 1;
    }

    puts("Connected\n");

    string mesAux = "list";
    strcpy(message, mesAux);

    // Send some data
    if (send(sock, message, strlen(message), 0) < 0)
    {
        puts("Send failed");
        return 1;
    }
    else
    {
        puts("send ok");
    }

    // Receive a reply from the server
    memset(server_reply, 0, 2000);
    if (recv(sock, server_reply, 2000, 0) < 0)
    {
        puts("recv failed");
        break;
    }
    else
    {
        puts("recv ok");
    }

    puts("Server reply :");
    puts(server_reply);

    close(sock);
    return 0;
}